#! /usr/bin/env python

# Mutual camera localization configuration file

PACKAGE='dot_finder'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name             Type      Level Description             Default Min   Max
# Blue
gen.add("BlueHueLow",  int_t,   0,   "Low hue value ",	  32,    0,     179)
gen.add("BlueHueHigh", int_t,   0,   "High hue value",	  158,     0,     179)

gen.add("BlueSatLow",  int_t,   0,   "Low saturation value",   64,     0,     255)
gen.add("BlueSatHigh", int_t,   0,   "High saturation value",  248,    0,     255)

gen.add("BlueValueLow",  int_t,   0,   "Low Value",		  131,    0,     255)
gen.add("BlueValueHigh", int_t,   0,   "High Value",		  255,    0,     255)

# Orange
gen.add("OrangeHueLow",  int_t,   0,   "Low hue value",	  151,    0,     179)
gen.add("OrangeHueHigh", int_t,   0,   "High hue value",	  33,     0,     179)

gen.add("OrangeSatLow",  int_t,   0,   "Low saturation value",   29,     0,     255)
gen.add("OrangeSatHigh", int_t,   0,   "High saturation value",  231,    0,     255)

gen.add("OrangeValueLow",  int_t,   0,   "Low Value",		  170,  0,     255)
gen.add("OrangeValueHigh", int_t,   0,   "High Value",		  255,  0,     255)

gen.add("maskToggle", bool_t,   0,    "MaskToggle", False)
gen.add("infoToggle", bool_t,   0,    "infoToggle", True)

gen.add("min_radius", int_t,   0,    "Smallest radius", 10, 0, 20)
gen.add("dilation_size", double_t,   0,    "dilation_size", 0.8, 0, 3)
gen.add("erosion_size", double_t,   0,    "erosion_size", 0.2, 0, 3)
gen.add("morph_type", int_t,   0,    "Smallest radius", 2, 0, 2)

gen.add("gaussian_sigma", double_t,   0,    "The standard deviation of the Gaussian blurring filter", 0.1, 0, 6)
gen.add("min_blob_area", double_t,   0,    "Smallest blob area found", 100,     0,     300)
gen.add("max_blob_area", double_t,   0,    "Largest blob area found", 2000,     0,     4000)
gen.add("max_width_height_distortion", double_t, 0, "Max height width distortion of blob", 0.6, 0, 1)
gen.add("max_circular_distortion", double_t, 0, "Max deviation from roundness", 0.8, 0, 10)

gen.add("line_angle_tolerance", double_t, 1, "Tolerance on the line angle between the two line", 0.05, 0, 1)
gen.add("hor_line_angle", double_t, 1, "Max horizontal angle of the cube", 0.35, 0, 1)
gen.add("pos_ratio", double_t, 1, "Positional ratio of the three LED", 3, 0, 4)
gen.add("pos_ratio_tolerance", double_t, 1, "Tolerance on the positional ratio", 0.6, 0, 1)
gen.add("radius_ratio_tolerance", double_t, 1, "Tolerance on the radius ratio", 0.3, 0, 1)
gen.add("ratio_int_tolerance", double_t, 1, "Tolerance on the ratio intensity of the led", 0.5, 0, 1.5)
gen.add("ratio_ellipse_max", double_t, 1, "Max ellipse ratio", 1.5, 0, 3)
gen.add("ratio_ellipse_min", double_t, 1, "Min ellipse ratio", 0.65, 0, 3)

gen.add("pos_left_led_cam_a", double_t, 0, "Position of the led on the left side of the camera A(Always positive)", 0.13, 0, 2)
gen.add("pos_right_led_cam_a", double_t, 0, "Position of the led on the right side of the camera A(Always positive)", 0.14, 0, 2)
gen.add("pos_left_led_cam_b", double_t, 0, "Position of the led on the left side of the camera B(Always positive)", 0.11, 0, 2)
gen.add("pos_right_led_cam_b", double_t, 0, "Position of the led on the right side of the camera B(Always positive)", 0.125, 0, 2)


exit(gen.generate(PACKAGE, "dot_finder", "DotFinder"))
